test/framework/fileConversion.go

```diff
@@ -1,35 +1,73 @@
 package framework
 
 import (
-	"os"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+
+	"github.com/kubevirt/containerized-data-importer/pkg/image"
 )
 
-const BaseImageFile = "../image/tinyCore.iso" // relative path to image file
+// createGz compresses a given test file and stores it to the target
+func CreateGz(src []byte) ([]byte, error) {
 
-type imageFile struct {
-	file   *os.File
-	name   string
-	size   uint64
-	md5Sum string
-}
+	var buf bytes.Buffer
+	gzw := gzip.NewWriter(&buf)
+	defer gzw.Close()
 
-type FileConverter struct {}
+	n, err := gzw.Write(src)
+	if n != len(src) {
+		return nil, fmt.Errorf("CreateGz: source size (%d) does not match bytes written (%d)\n", n, len(src))
+	}
+	if err != nil {
+		return nil, fmt.Errorf("CreateGz: %v\n", err)
+	}
 
-// createGz compresses a given test file and stores it to the target
-func (f *FileConverter) CreateGz() {}
+	return buf.Bytes(), nil
+}
 
-func (f *FileConverter) CreateXz() {}
+func CreateXz() {}
 
-func (f *FileConverter) CreateTar() {}
+func CreateTar() {}
 
-func (f *FileConverter) CreateTarGz() {
-	f.CreateTar()
-	f.CreateTarGz()
+func CreateTarGz() {
+	CreateTar()
+	CreateTarGz()
 }
 
-func (f *FileConverter) CreateTarXz() {
-	f.CreateTar()
-	f.CreateTarXz()
+func CreateTarXz() {
+	CreateTar()
+	CreateTarXz()
 }
 
-func (f *FileConverter) CreateQcow2() {}
+func CreateQcow2() {}
+
+func FormatTestData(src []byte, targetFileName string, targetFormat image.FileExt) ([]byte, error) {
```

Comment:
Just fyi, `targetFileName` is not referenced.

Reply:
Thanks, removed.  Initially intended to write to disk but it's not needed for the moment.